src/render.rs

```diff
@@ -34,65 +104,152 @@ fn rgb(r: u8, g: u8, b: u8) -> [TmdsPair; 3] {
 
 impl ScanRender {
     pub fn new() -> Self {
-        let mut sb = ScanlistBuilder::new(640, 480 / VERTICAL_REPEAT as u32);
-        sb.begin_stripe(320 / VERTICAL_REPEAT as u32);
-        sb.solid(92, rgb(0xc0, 0xc0, 0xc0));
-        sb.solid(90, rgb(0xc0, 0xc0, 0));
-        sb.solid(92, rgb(0, 0xc0, 0xc0));
-        sb.solid(92, rgb(0, 0xc0, 0x0));
-        sb.solid(92, rgb(0xc0, 0, 0xc0));
-        sb.solid(90, rgb(0xc0, 0, 0));
-        sb.solid(92, rgb(0, 0, 0xc0));
-        sb.end_stripe();
-        sb.begin_stripe(40 / VERTICAL_REPEAT as u32);
-        sb.solid(92, rgb(0, 0, 0xc0));
-        sb.solid(90, rgb(0x13, 0x13, 0x13));
-        sb.solid(92, rgb(0xc0, 0, 0xc0));
-        sb.solid(92, rgb(0x13, 0x13, 0x13));
-        sb.solid(92, rgb(0, 0xc0, 0xc0));
-        sb.solid(90, rgb(0x13, 0x13, 0x13));
-        sb.solid(92, rgb(0xc0, 0xc0, 0xc0));
-        sb.end_stripe();
-        sb.begin_stripe(120 / VERTICAL_REPEAT as u32);
-        sb.solid(114, rgb(0, 0x21, 0x4c));
-        sb.solid(114, rgb(0xff, 0xff, 0xff));
-        sb.solid(114, rgb(0x32, 0, 0x6a));
-        sb.solid(116, rgb(0x13, 0x13, 0x13));
-        sb.solid(30, rgb(0x09, 0x09, 0x09));
-        sb.solid(30, rgb(0x13, 0x13, 0x13));
-        sb.solid(30, rgb(0x1d, 0x1d, 0x1d));
-        sb.solid(92, rgb(0x13, 0x13, 0x13));
-        sb.end_stripe();
-        let scanlist = sb.build();
         let stripe_remaining = 0;
         let scan_ptr = core::ptr::null();
         let scan_next = core::ptr::null();
+        // Safety: it makes sense for two cores to both have access to the
+        // fifo, as it's designed for that purpose. A better PAC API might
+        // allow us to express this safely.
+        let pac = unsafe { pac::Peripherals::steal() };
```

Comment:
It may be worth switching to the chiptool based pac given how much we are having to force this pac to our will. But I think we can sort of sidestep the `Peripherals` struct in certan parts and just construct the [`*_SPEC`](https://docs.rs/rp2040-pac/latest/rp2040_pac/index.html?search=_SPEC) structs directly as they just look to be zero-sized types ([example](https://docs.rs/rp2040-pac/latest/rp2040_pac/dma/ch/ch_read_addr/struct.CH_READ_ADDR_SPEC.html)) So when we need just one register we can use this.